### PR TITLE
Show 2 sigfigs for numbers >= 1 and 1 sigfig for numbers < 1.

### PR DESCRIPTION
**Pull request by jy2wong**

### CODE REVIEW

**Comment by jy2wong** [comment]
Note: this spectacularly breaks any language other than english right now. Merging this requires careful coordination.

**Review by gyp — APPROVED** [review]

File: src/components/calculator/util/FormatPrecision.ts [hunk 1]
```diff
@@ -1,115 +1,63 @@
...
- * This is necessary because float.toPrecision will use exponential notation for large or small numbers.
+ * By default, show 2 sigfigs for numbers greater than 1 and 1 sigfig for
+ * numbers less than 1.
  */
 export function fixedPointPrecision(
   valOrNull: number | null,
   maxNumber = 1e6,
-  decimalsNearMax = false,
-  thousandsSeparator = ',',
+  minNumber = 1,
```
Comment (beshaya) [inline src/components/calculator/util/FormatPrecision.ts:9]: This variable name seems misleading to me. `maxNumber` is the largest possible number microCOVID will ever display. `minNumber` appears to be the smallest number for which 2 decimal places will be shown?

Perhaps minNumberForTwoSigFigs? Or could we just always show two sigfigs?

File: src/components/calculator/util/FormatPrecision.ts [hunk 2]
```diff
@@ -1,115 +1,63 @@
...
-    return (val * 100).toFixed() + suffix
-  }
-  return (
-    fixedPointPrecision(
-      val * 100,
-      /*maxNumber=*/ 100,
-      /*decimalsNearMax=*/ true,
-      /*thousandsSeparator=*/ '',
-    ) + suffix
+export function fixedPointPrecisionPercent(val: number | null): string {
```
Comment (beshaya) [inline src/components/calculator/util/FormatPrecision.ts:56]: I don't understand how this works - don't we need to multiply by 100 somewhere?

File: src/components/calculator/util/budgetUtil.ts [hunk 3]
```diff
@@ -1,31 +1,59 @@
...
-  percentage_suffix: string,
-): string => {
+): BudgetConsumptionInfo => {
   const displayedPoints = Number(
     fixedPointPrecision(
       points,
       /*maxNumber=*/ undefined,
-      /*decimalsNearMax=*/ undefined,
-      /*thousandsSeparator=*/ '',
+      /*minNumber=*/ undefined,
```
Comment (beshaya) [inline src/components/calculator/util/budgetUtil.ts:26]: Could we use an object parameter instead? https://exploringjs.com/impatient-js/ch_callables.html#named-parameters

File: src/components/calculator/util/FormatPrecision.ts (hunk 2 above)
Comment (jy2wong) [reply src/components/calculator/util/FormatPrecision.ts:56]: `numberStyle=percent` handles it.

**Comment by jy2wong** [comment]
Remaining items:
- update weblate translations manually
- submit!

File: src/components/calculator/util/budgetUtil.ts (hunk 3 above)
Comment (jy2wong) [reply src/components/calculator/util/budgetUtil.ts]: done

File: src/components/calculator/util/FormatPrecision.ts (hunk 1 above)
Comment (jy2wong) [reply src/components/calculator/util/FormatPrecision.ts]: renamed minNumber -> oneSigFigThreshold.

**Comment by blanchardjeremy** [comment]
Very excited to see this live. Any way we can support? (I don't know how to do the weblate stuff. We should probably document that so we all know how.)

**Pull request merged**